bson/src/main/org/bson/BsonBinaryWriter.java

```diff
@@ -415,7 +415,7 @@ private void backpatchSize() {
 
     private void validateSize(final int size) {
         if (size > maxDocumentSizeStack.peek()) {
-            throw new BsonSerializationException(format("Document size of %d is larger than maximum of %d.", size,
+            throw new BsonMaximumSizeExceededException(format("Document size of %d is larger than maximum of %d.", size,
```

Comment:
Looks like this same change should be made in com.mongodb.connection.BsonWriterHelper#writePayload

Reply:
Good catch.